### PR TITLE
Acknowledgement updated for reset

### PR DESCRIPTION
**Pull request by Shevil27**
## Description
I have added a few more react components inside the acknowledgement file.

## Implementation
- If the reset password link's token is valid, then it will move forward, else it wil show a 404 error.
- Instead of hardcoded `href`, used dynamic href hook.

### CODE REVIEW

File: src/public/view/Reset/Forgot.tsx [hunk 1]
```diff
@@ -6,8 +6,10 @@ import ky from 'ky';
 
 import { AuthView } from '../../component/AuthView';
 import { rootRoute } from '../../Root';
-import { EmailSent } from './EmailSent';
+import { useHref } from '../../../util/location';
+import { EmailSent } from './acknowledgement';
```
Comment (ubcharu) [inline src/public/view/Reset/Forgot.tsx]: Components containing React components is always `UpperCamelCase`.

File: src/public/view/Reset/Forgot.tsx [hunk 2]
```diff
@@ -6,8 +6,10 @@ import ky from 'ky';
 
 import { AuthView } from '../../component/AuthView';
 import { rootRoute } from '../../Root';
-import { EmailSent } from './EmailSent';
+import { useHref } from '../../../util/location';
```
Comment (ubcharu) [inline src/public/view/Reset/Forgot.tsx:10]: Import order. Parent component first and then all the way down to the current working directory.

File: src/public/view/Reset/Reset.tsx [hunk 3]
```diff
@@ -27,9 +32,8 @@ async function resetPassword(payload: ResetCredentials): Promise<any> {
 }
 
 async function getTokenInfo(token: string): Promise<any> {
-  const response = await ky
-    .get(`/auth/reset-password/${token}`)
-    .json();
+  const response = await ky.get(`/auth/reset-password/${token}`).json();
+  console.log(response);
```
Comment (ubcharu) [inline src/public/view/Reset/Reset.tsx]: Remove `console.log`.

File: src/public/view/Reset/Reset.tsx [hunk 4]
```diff
@@ -70,13 +74,19 @@ export function Reset() {
         <Heading level={1} alignSelf='self-start'>
           Reset Password
         </Heading>
-        <ResetForm
-          token={resetToken}
-          inProgress={reset.isPending}
-          onSubmit={reset.mutate}
-        />
+        {info.isPending ? (
```
Comment (ubcharu) [inline src/public/view/Reset/Reset.tsx]: This is wrong. You should show form when info is successful and when submit is successful i.e. `reset.isSuccessful`, then show `ResetPasswordSuccessful` component.

Looks like, the conditions are not tested.

File: src/public/view/Reset/acknowledgement.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+export function EmailSent() {
+  return (
+    <Heading level={4} alignSelf={'center'}>
+      Link to reset your password has been sent to you email. Please check.
+    </Heading>
+  );
+}
+
+export function ResetPasswordSuccessfull() {
```
Comment (ubcharu) [inline src/public/view/Reset/acknowledgement.tsx]: Name should be `ResetPasswordSuccessful` and not `ResetPasswordSuccessfull`.

File: src/public/view/Reset/acknowledgement.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    </Heading>
+  );
+}
+
+export function ResetPasswordFailed() {
+  return (
+    <Heading level={1} alignSelf={'center'}>
+      404
+      <AlertCircleFilled />
+      <Heading level={4}>Reset password link not found</Heading>
```
Comment (ubcharu) [inline src/public/view/Reset/acknowledgement.tsx]: You cannot have heading inside a heading. That will result in bad HTML markup.

**Review by ubcharu — CHANGES_REQUESTED** [review]
@Shevil27, Please make necessary changes.

File: src/public/view/Reset/Reset.tsx [hunk 7]
```diff
@@ -51,12 +47,20 @@ function useResetPassword() {
...
 
+type State = 'ValidToken' | 'InvalidToken' | 'ResetSuccess' | 'ResetFailure';
+
 export function Reset() {
   const { resetToken } = resetRoute.useParams();
   const info = useTokenInfo(resetToken);
   const reset = useResetPassword();
   const loginHref = useHref(loginRoute);
 
+  const tokenState: State = info.isSuccess ? 'ValidToken' : 'InvalidToken';
```
Comment (ubcharu) [inline src/public/view/Reset/Reset.tsx]: The whole idea of creating `State` type is such that your application remains in exactly one state. If you have two state variables, then it defeats the purpose of create single state value.

File: src/public/view/Reset/Reset.tsx [hunk 8]
```diff
@@ -74,17 +78,15 @@ export function Reset() {
         <Heading level={1} alignSelf='self-start'>
           Reset Password
         </Heading>
-        {info.isPending ? (
+        {tokenState === 'ValidToken' && resetState === 'ResetFailure' && (
```
Comment (ubcharu) [inline src/public/view/Reset/Reset.tsx]: This multiple conditions should not happen. Only single state check should be there.

File: src/public/view/Reset/Reset.tsx [hunk 9]
```diff
@@ -74,17 +78,15 @@ export function Reset() {
...
             token={resetToken}
             inProgress={reset.isPending}
             onSubmit={reset.mutate}
           />
-        ) : info.isSuccess ? (
-          <ResetPasswordSuccessfull />
-        ) : (
-          <ResetPasswordFailed />
         )}
+        {resetState === 'ResetSuccess' && <ResetPasswordSuccessfull />}
```
Comment (ubcharu) [inline src/public/view/Reset/Reset.tsx]: Component name should be `ResetPasswordSuccessful`!

**Review by ubcharu — CHANGES_REQUESTED** [review]
Some more review comments!

**Review by mistyharsh — APPROVED** [review]

**Pull request merged**